### DISCUSSION TITLE
Help with basic example

### DISCUSSION THREAD

**Discussion by Gareth064** [microsoft/RulesEngine#514] [Q&A]
Hey guys. I have just picked up this package as it looks like it will be very useful when applying Rules to make UI changes on my Blazor WASM App. But before I start adding it in, I am doing a small test app to see how it works.... and I'm already stumped :(

Can anyone tell me what is wrong here...

```cs
//{
//    "WorkflowName": "Discount",
//    "Rules": [
//      {
//        "RuleName": "GiveDiscount10",
//        "Expression": "input1.country == \"india\""
//      }
//    ]
//  }

string[] workflowRules = { "  {\r\n    \"WorkflowName\": \"CountryCheck\",\r\n    \"Rules\": [\r\n      {\r\n        \"RuleName\": \"IsIndia\",\r\n        \"Expression\": \"input1.country == \\\"india\\\"\"\r\n      }\r\n    ]\r\n  }" };
var re = new RulesEngine.RulesEngine(workflowRules);
var input1 = new input1 { country = "india" };
var resultList = await re.ExecuteAllRulesAsync("CountryCheck", input1);

foreach (var result in resultList)
{
    if (result.IsSuccess)
    {
        Console.WriteLine($"Rule - {result.Rule.RuleName}, IsSuccess - {result.IsSuccess}");
    }
    else
    {
        Console.WriteLine($"Rule - {result.Rule.RuleName}, ExceptionMessage - {result.ExceptionMessage}");
    }
}

public class input1
{
    public string country { get; set; }
}
```

Results in
```
Rule - IsIndia, ExceptionMessage - Exception while parsing expression `input1.country == "india"` - No property or field 'country' exists in type 'input1'
```

**Comment by abbasc52** [comment]
Hi @Gareth064 ,

There is a bug in our dependency library where if a class has same name as input parameter, it gives higher precedence to type.
While I am working towards resolving this, you can use the below workaround:- 

You can either set AutoRegisterInputType as false in ReSettings
`var re = new RulesEngine.RulesEngine(workflowRules,  new ReSettings {AutoRegisterInputType : false });`

Or you can change the name of your input parameter

```c#
//{
//    "WorkflowName": "Discount",
//    "Rules": [
//      {
//        "RuleName": "GiveDiscount10",
//        "Expression": "myInput.country == \"india\""
//      }
//    ]
//  }

var re = new RulesEngine.RulesEngine(workflowRules);
var input1 = new input1 { country = "india" };
var resultList = await re.ExecuteAllRulesAsync("CountryCheck", new RuleParameter("myInput",input1);

```

**Comment by Gareth064** [comment]
Awesome, thank you @abbasc52 this is working and getting me going the way I was hoping.
I only need this for some basic stuff to begin with, like setting a CSS value on a Blazor Component depending on values of the data bound to it. And the reason I need something like this library, is because I want the user of my app to define the logic themselves.
